granulate_utils/metrics/yarn/utils.py

```diff
@@ -160,6 +162,14 @@ def detect_yarn_config(*, logger: Union[logging.Logger, logging.LoggerAdapter])
         logger.debug(f"found YARN home dir: {yarn_home_dir}")
         yarn_site_xml_file = Path(yarn_home_dir).joinpath(RELATIVE_YARN_SITE_XML_PATH)
         return read_config_file(yarn_site_xml_file, logger=logger)
+
+    # try to find YARN config in environment variable
+    else:
```

Comment:
This is not enough - the `yarn_home_dir` might exist but the config file does not. Change the code such that is checks if `yarn_site_xml_file` exists before reading, and resorts to the `YARN_CONF_DIR_ENV_VAR` if missing.

Reply:
Why checking if exists? it happens inside `read_config_file`, We also not check if the file exists few lines above